src/middleware/actix3.rs

```diff
@@ -0,0 +1,95 @@
+mod request_response_impl;
+
+pub mod request;
+pub mod response;
+
+use actix_http::http::HeaderName;
+use std::collections::HashMap;
+use tokio02::sync::mpsc::Receiver;
+
+use crate::{
+    generic_http::{GenericRequest, GenericResponse},
+    har_builder::HarBuilder,
+    SpeakeasySdk,
+};
+
+pub(crate) fn speakeasy_header_name() -> HeaderName {
+    HeaderName::from_static("speakeasy-request-id")
+}
+
+#[derive(Debug)]
+pub(crate) enum Message {
+    Request {
+        request_id: String,
+        request: GenericRequest,
+    },
+    Response {
+        request_id: String,
+        response: GenericResponse,
+    },
+}
+
+pub struct Middleware {
+    sdk: SpeakeasySdk,
+    requests: HashMap<String, GenericRequest>,
```

Comment:
This looks like state; is there a way to make this more state-less?

Reply:
I don't believe so, and if you look in the `sdk-controller` PR there is more state for custom-id, request specific masking, request specific path-hint. The request is captured before it is handed to the user, and the response is captured after the user has handled the request and send the response off. So I don't think we could avoid having that piece of state until the response is finished. What is your main concern with having state here?

Reply:
I think the concern is really more around having a singleton that's managing the state. The other sdks maintain state in a per request/response way, which feels a little safer. I think. The docs for the middleware that you linked above have an example of something that would work that way, I think.

Reply:
Ya that's true I think we could put everything in the controller using `ReqData`. And leave just the global default masking rules as the global data. That might make the code easier to understand too. Let me think if there's any reason that can't be done. If not I'll change it in a new PR (just because there's other PRs that builds on this so it would need to be changed in a few places).

Reply:
Yeah, trying to minimize global state sounds like a good plan

Reply:
Sounds good. Although with Rust and this access pattern (using messages) it’s not as big an anti pattern. But I agree I think it can be improved in the future. 

Putting it in the request will require an Arc and Mutex. But I think that still makes more sense.